src/sonic-yang-models/yang-models/sonic-cable-length.yang

```diff
@@ -0,0 +1,65 @@
+module sonic-cable-length {
+
+    yang-version 1.1;
+
+    namespace  "http://github.com/Azure/sonic-cable-length";
+
+    prefix cable-length;
+
+    import sonic-port {
+        prefix port;
+    }
+
+    description "CABLE_LENGTH YANG module for SONiC OS";
+
+    revision 2021-11-11 {
+        description "Initial version";
+    }
+
+    container sonic-cable-length {
+        container CABLE_LENGTH {
+
+            description "CABLE_LENGTH part of config_db.json";
+
+            list CABLE_LENGTH_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
```

Comment:
I think, this name is expected to be AZURE now, correct? if yes, let's just put what we're supporting now.

Reply:
I don’t tend to agree on that. IMO any name should be supported.

Reply:
don't see "AZURE" used anywhere in the backend except in the unit tests. will that be a problem if we allow any string?

Reply:
I see there isn't any restriction of the hash name in `buffermgr`:
https://github.com/Azure/sonic-swss/blob/d95823d057035e518c38cdea40e3a330e4bb9255/cfgmgr/buffermgr.cpp#L384-L388

Reply:
I think the logic in the code should be to load the only item in `CABLE_LENGTH` table regardless of its name.
In the vs test, it is assumed that the vs image is deployed via using default configuration rendered from templates in sonic-buildimage. In the templates, the names of the items are `AZURE`.
But a user can configure it to other names in his/her production environment.

Reply:
Ok, I've not seen any other name usage other than AZURE in the name, also can the CABLE_LENGTH_LIST list have more than one entry? please add the test cases with names other than AZURE.

Reply:
Just FYI, AFAIK it is assumed only one entry in the list and only the first one is handled.

Reply:
@venkatmahalingam, added unit test, please review, thanks!

Reply:
I've reviewed the new changes, I'm okay to approved as soon as the below comment is addressed.

Reply:
Could you please be more specific about which comment should be addressed? @venkatmahalingam

Reply:
I'm referring to the comment on the length field.

                    leaf length {
                        type string {
                            pattern '[0-9]+m' {
In the meanwhile, we need to put an upper bound on the cable length. I think ~10km (9999m) should be large enough for all scenarios.

Reply:
agreed that we don't need an upper bound now